saml2/saml2-service-provider/src/opensaml4Main/java/org/springframework/security/saml2/provider/service/authentication/OpenSaml4AuthenticationProvider.java

```diff
@@ -359,12 +359,26 @@ public void setResponseAuthenticationConverter(
 			Response response = responseToken.getResponse();
 			Saml2AuthenticationToken token = responseToken.getToken();
 			Saml2ResponseValidatorResult result = Saml2ResponseValidatorResult.success();
+			AbstractSaml2AuthenticationRequest request = token.getAuthenticationRequest();
 			String statusCode = getStatusCode(response);
 			if (!StatusCode.SUCCESS.equals(statusCode)) {
 				String message = String.format("Invalid status [%s] for SAML response [%s]", statusCode,
 						response.getID());
 				result = result.concat(new Saml2Error(Saml2ErrorCodes.INVALID_RESPONSE, message));
 			}
+			String inResponseTo = response.getInResponseTo();
```

Comment:
We probably should check the `inResponseTo` value in each assertion as well in `createAssertionValidator`.